src/io/common.zig

```diff
@@ -7,16 +7,19 @@ const assert = std.debug.assert;
 
 const is_linux = builtin.target.os.tag == .linux;
 
-pub const ListenOptions = struct {
-    rcvbuf: c_int,
-    sndbuf: c_int,
+pub const TCPOptions = struct {
+    rcvbuf: c_int = 0,
+    sndbuf: c_int = 0,
     keepalive: ?struct {
         keepidle: c_int,
         keepintvl: c_int,
         keepcnt: c_int,
-    },
-    user_timeout_ms: c_int,
-    nodelay: bool,
+    } = null,
+    user_timeout_ms: c_int = 0,
+    nodelay: bool = false,
+};
```

Comment:
FWIW, originally I didn't add defaults there by design, as per our 🐯-style:

> Explicitly pass options to library functions at the call site, instead of relying on the defaults. For example, write `@prefetch(a, .{ .cache = .data, .rw = .read, .locality = 3 });` over `@prefetch(a, .{});`. This improves readability but most of all avoids latent, potentially catastrophic bugs in case the library ever changes its defaults.

For tests, we can have `count test_options: TCPOptions = ...` to avoid verbosity.